godot-core/src/builtin/plane.rs

```diff
@@ -191,6 +310,301 @@ mod test {
+        assert_eq!(plane.intersects_segment(vec_d, vec_c), Some(vec_a));
+        assert_eq!(plane.intersects_segment(vec_d, vec_e), None);
+        // From a point straight up the origin point, the distance from the origin to the intersection point
+        // formed by a slanted segment should follow the Pythagorean Theorem only if the segment ended on or
+        // beyond the `plane`.
+        assert_eq!(
+            plane.intersects_segment(vec_c, vec_f).unwrap().length(),
+            (vec_c.distance_squared_to(vec_f) - vec_c.length_squared()).sqrt()
+        );
+        assert_eq!(
+            plane.intersects_segment(vec_b, vec_f).unwrap().length(),
+            (vec_b.distance_squared_to(vec_f) - vec_b.length_squared()).sqrt()
+        );
+        // A segment parallel to the `plane` would not intersect the plane.
+        assert_eq!(plane.intersects_segment(vec_c, vec_g), None);
+        // A segment pointing to the opposite direction as the `plane` would not intersect it.
+        assert_eq!(plane.intersects_segment(vec_b, vec_c), None);
+        assert_eq!(plane.intersects_segment(vec_b, vec_g), None);
+    }
+
+    /// Tests `is_equal_approx()`.
+    #[test]
+    fn test_equal() {
+        // Initial normalized planes.
+        let plane_a: Plane = Plane::new(Vector3::new(0.0, 0.0, 1.0).normalized(), 0.0);
+        let plane_b: Plane = Plane::new(Vector3::new(0.01, 0.0, 1.0).normalized(), 0.0);
+        let plane_c: Plane = Plane::new(Vector3::new(0.0001, 0.0, 1.0).normalized(), 0.0);
+        let plane_d: Plane = Plane::new(Vector3::new(0.000001, 0.0, 1.0).normalized(), 0.0);
+        let plane_e: Plane = Plane::new(Vector3::new(0.000001, 0.0, 1.0).normalized(), 0.01);
+        let plane_f: Plane = Plane::new(Vector3::new(0.000001, 0.0, 1.0).normalized(), 0.000001);
+
+        // Same planes should be equals.
+        assert_eq_approx!(&plane_a, &plane_a, Plane::is_equal_approx);
+        assert_eq_approx!(&plane_b, &plane_b, Plane::is_equal_approx);
+        // Although similar, planes below shouldn't be approximately equals.
+        assert_ne_approx!(&plane_a, &plane_b, Plane::is_equal_approx);
+        assert_ne_approx!(&plane_a, &plane_c, Plane::is_equal_approx);
+        // Planes below should be approximately equal because it's lower than the set tolerance constant.
+        assert_eq_approx!(&plane_a, &plane_d, Plane::is_equal_approx);
+        // Although approximately equal in the `normal` part, it is not approximately equal in the `d`
+        // part.
+        assert_ne_approx!(&plane_a, &plane_e, Plane::is_equal_approx);
+        // Both attributes are approximately equal.
+        assert_eq_approx!(&plane_a, &plane_f, Plane::is_equal_approx);
+        // Although considered approximately equal with `plane_a`, `plane_b` is not considered approximately
+        // equal with `plane_e` because the baseline comparison is tighter.
+        assert_ne_approx!(&plane_b, &plane_e, Plane::is_equal_approx);
```

Comment:
Really appreciate the comments explaining why or why not approximate equality applies! 👍 

Could you group them to have first all the equality checks, then the non-equality ones? Also please add an empty line before a new `///` comment block, to make it immediately clear that a comment refers to the code below it.